integration/test/performance_tracker/transmitter/transmitter.go

```diff
@@ -0,0 +1,236 @@
+*/
+func (transmitter * TransmitterAPI) TableExist() (bool,error){
+	// l,err := transmitter.ListTables()
+	// for i:=0; i< len(l); i++{
+	// 	if transmitter.DataBaseName == l[i]{
+	// 		return true,nil
+	// 	}
+	// }
+	// return false,err
+	exists := true
+	_, err := transmitter.dynamoDbClient.DescribeTable(
+		context.TODO(), &dynamodb.DescribeTableInput{TableName: aws.String(transmitter.DataBaseName)},
+	)
+	if err != nil {
+		var notFoundEx *types.ResourceNotFoundException
+		if errors.As(err, &notFoundEx) {
+			fmt.Printf("Table %v does not exist.\n", transmitter.DataBaseName)
+			err = nil
+		} else {
+			fmt.Printf("Couldn't determine existence of table %v. Here's why: %v\n",transmitter.DataBaseName, err)
+		}
+		exists = false
+	}
+	return exists, err
+}
+/*
+RemoveTable()
+Desc: Removes the table that was craeted with initialization. 
+*/
+func (transmitter * TransmitterAPI) RemoveTable() error{
+	input := dynamodb.DeleteTableInput{TableName: aws.String(transmitter.DataBaseName)}
+
+	_,err:=transmitter.dynamoDbClient.DeleteTable(&input)
+	if err !=nil{
+		fmt.Println(err)
+	}
+	return err
+}
+// ListTables lists the DynamoDB table names for the current account.
+func (transmitter * TransmitterAPI) ListTables() ([]string, error) {
+	var tableNames []string
+	input := &dynamodb.ListTablesInput{}
+	tables, err := transmitter.dynamoDbClient.ListTables(
+		input)
+	if err != nil {
+		fmt.Printf("Error listing tables: %v\n", err)
+	} else {
+		for  i:=0; i< len(tables.TableNames); i++{
+			tableNames =  append(tableNames,*tables.TableNames[i])
+		}
+		
+	}
+	return tableNames, err
+}
+/*
+SendItem()
+Desc: Parses the input data and adds it to the dynamo table
+Param: data []byte is the data collected by data collector 
+*/
+func (transmitter * TransmitterAPI) SendItem(data []byte) (string,error) {
+	// return nil
+	packet, err := transmitter.Parser(data)
+	if err != nil{
+		return "",err
+	}
+	sentItem,err := transmitter.AddItem(packet) 
+	return sentItem,err
+}
+
+func (transmitter * TransmitterAPI) Parser(data []byte) (map[string]interface{},error){
+	dataHolder := collectorData{}
+	err := json.Unmarshal(data,&dataHolder)
+	if err !=nil{
+		return nil,err
+	}
+	packet := make(map[string]interface{})
+	//@TODO: add git integration temp solution
+	packet["Hash"] = fmt.Sprintf("%d",time.Now().UnixNano())
+	packet["CommitDate"] = fmt.Sprintf("%d",time.Now().UnixNano())
+	/// will remove
+	for _,rawMetricData:= range dataHolder{
+		numDataPoints := float64(len(rawMetricData.Timestamps))
+		// @TODO: Export this part to GetMetricStatistics after merging with data collection code
+		sum :=0.0
+		for _,val := range rawMetricData.Values {
+			sum += val
+		}
+		avg :=  sum /numDataPoints
+		// calculate diff between mean and values 
+		diffSum := 0.0
+		for _,val := range rawMetricData.Values{
+			diffSum = diffSum +  (avg - val)
+		}
+		//----------------
+		metric := Metric{
```

Comment:
Maybe I'm wrong. @SaxyPandaBear do we already have the aggregator plugins that gave us more statistical values? So why do we want to aggregate the "aggregated" metrics again?

Reply:
I'm not sure what you're referring to. This isn't running inside the agent. This is part of the performance test. What this is doing is expecting a bunch of raw values and then is doing the calculations in memory. What we want to do is move away from this to just call GetMetricStatistics to cut out a lot of this work.